File: app/src/main/java/igrek/songbook/system/PackageInfoService.java
package igrek.songbook.system;

import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.support.v7.app.AppCompatActivity;

import javax.inject.Inject;

import igrek.songbook.dagger.DaggerIoc;
import igrek.songbook.info.logger.Logger;
import igrek.songbook.info.logger.LoggerFactory;

public class PackageInfoService {
	
	@Inject
	AppCompatActivity activity;
	
	private Logger logger = LoggerFactory.INSTANCE.getLogger();
	private String versionName;
	private int versionCode;
	
	public PackageInfoService() {
		DaggerIoc.getFactoryComponent().inject(this);
		
		try {
			PackageInfo pInfo = activity.getPackageManager()
					.getPackageInfo(activity.getPackageName(), 0);
			versionName = pInfo.versionName;
			versionCode = pInfo.versionCode;
		} catch (PackageManager.NameNotFoundException e) {
			logger.error(e);
		}
	}
	
	public String getVersionName() {
		return versionName;
	}
	
	public int getVersionCode() {
		return versionCode;
	}
}

File: app/src/main/java/igrek/songbook/system/PackageInfoService.kt
package igrek.songbook.system

import android.content.pm.PackageManager
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import igrek.songbook.info.logger.LoggerFactory
import igrek.songbook.inject.LazyExtractor
import igrek.songbook.inject.LazyInject
import igrek.songbook.inject.appFactory

@Suppress("DEPRECATION")
class PackageInfoService(
    appCompatActivity: LazyInject<AppCompatActivity> = appFactory.appCompatActivity,
) {
    private val activity by LazyExtractor(appCompatActivity)

    private val logger = LoggerFactory.logger
    var versionName: String? = null
        private set
    var versionCode: Long = 0
        private set

    init {
        try {
            val pInfo = activity.packageManager
                .getPackageInfo(activity.packageName, 0)
            versionName = pInfo.versionName
            versionCode = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
                pInfo.longVersionCode
            } else {
                pInfo.versionCode.toLong()
            }
        } catch (e: PackageManager.NameNotFoundException) {
            logger.error(e)
        }
    }
}
